audit_log/utils.py

```diff
@@ -1,10 +1,15 @@
+import threading
+from typing import Union
+
 from django.conf import settings
 from django.utils import timezone
 
 from audit_log.enums import Operation, Role, Status
 from audit_log.models import AuditLogEntry
 from events.auth import ApiKeyUser
 
+_THREAD_LOCALS = threading.local()
```

Comment:
I don't think that you have missed anything. The usage of either the view or the request for the IDs is a good point. I've now implemented this so that the `_audit_logged_object_ids` attribute is set on the request.